### PR TITLE
Adds overload to restrict dispatch access to the view function.

### PR DESCRIPTION
**Pull request by pardom**

### CODE REVIEW

**Comment by noheltcj** [comment]
This looks good to me. Tested the ergonomics of the change; very much improves the experience. Of course there are pros/cons to deprecating the option to have dispatch in the render function. Outside of situations where users wrap the runtime to synchronously obtain the first set of props, this new method has no known drawbacks. But, in cases where synchronous initial props are desired, having the dispatch be potentially nullable is useful (or at least having the choice to do so is useful) to avoid essentially rebuilding the runtime or practically doing so in a wrapper. For these users to keep the props-only render function pattern, they must move the decision to handle null or crash to the wrapper or build a custom mechanism over Oolong instead of simply providing a nullable dispatch and checking that when a props-provided event would be triggered.

Outside of that edge case, the new API improves ergonomics and should be the recommended way to use Oolong in my opinion.

**Comment by dallasgutauckis** [comment]
I think this should also include changes to the `core-concepts.md` file to remove dispatch from props function examples and move that into the view implementation

File: oolong/src/commonMain/kotlin/oolong/runtime.kt [hunk 1]
```diff
@@ -1,55 +1,93 @@
...
-import kotlinx.coroutines.launch
 import kotlin.coroutines.CoroutineContext
 import kotlin.jvm.JvmOverloads
+import kotlinx.coroutines.*
 
-/**
- * Create a runtime.
- */
+internal const val DEPRECATION_MESSAGE_DISPATCH_RELOCATED =
+    "The dispatch function should now be accessed via the view function. It will be removed from the props function in a future release."
```
Comment (dallasgutauckis) [inline oolong/src/commonMain/kotlin/oolong/runtime.kt:8]: props -> render

File: docs/index.md [hunk 2]
```diff
@@ -36,16 +36,16 @@ val init: () -> Pair<Model, Effect<Msg>> = {
 
 val update: (Msg, Model) -> Pair<Model, Effect<Msg>> = { msg, model ->
     when (msg) {
-        Msg.Increment -> model.copy(count = model.count + 1)
-        Msg.Decrement -> model.copy(count = model.count - 1)
-    } to none()
+        Msg.Increment -> netxt(model.copy(count = model.count + 1))
+        Msg.Decrement -> netxt(model.copy(count = model.count - 1))
```
Comment (dallasgutauckis) [inline docs/index.md:40]: typo: netxt -> next

File: docs/index.md [hunk 3]
```diff
@@ -36,16 +36,16 @@ val init: () -> Pair<Model, Effect<Msg>> = {
...
-        Msg.Increment -> model.copy(count = model.count + 1)
-        Msg.Decrement -> model.copy(count = model.count - 1)
-    } to none()
+        Msg.Increment -> next(model.copy(count = model.count + 1))
+        Msg.Decrement -> next(model.copy(count = model.count - 1))
+    }
 }
 
-val view: (Model) -> Props = { model ->
+val view: (Model, dispatch) -> Props = { model ->
```
Comment (dallasgutauckis) [inline docs/index.md:44]: this line should be:

```
val view: (Model, Dispatch<Msg>) -> Props = { model, dispatch ->
```

File: docs/index.md [hunk 4]
```diff
@@ -99,29 +100,29 @@ val init: () -> Pair<Model, Effect<Msg>> = {
...
+        { dispatch(Msg.Increment) },
+        { dispatch(Msg.Decrement) }
     )
 }
 
 fun main() {
     runBlocking {
 //sampleStart
-        val render: (Props, Dispatch<Msg>) -> Any? = { props, dispatch ->
+        val render: (Props) -> Any? = { props, dispatch ->
```
Comment (dallasgutauckis) [inline docs/index.md:119]: remove `, dispatch`

**Review by dallasgutauckis — APPROVED** [review]

**Pull request merged**